### DISCUSSION TITLE
Why is my body falling through the ground when mass=1 && intertia = identity?

### DISCUSSION THREAD

**Discussion by Pindrought** [jrouwe/JoltPhysics#733] [Q&A] [answered]
**Edit: I see now that all of the 3d convex hull shapes are using CalculateMassAndInertia. I had copied the bit from the 2d shape generation, and I guess that is the real issue here.**
-----------------------------------------------------------------

I am creating a StaticCompoundShape consisting of convex hulls.

If I use the following, my shape easily falls through the ground.
```cpp
creation_settings.mOverrideMassProperties = EOverrideMassProperties::MassAndInertiaProvided;
creation_settings.mMassPropertiesOverride.mMass = 1.0f;
creation_settings.mMassPropertiesOverride.mInertia = Mat44::sIdentity();
```

**If I use the auto calculate mass and inertia, I do not have any issues with the shape falling through the ground.**
```cpp
creation_settings.mOverrideMassProperties = EOverrideMassProperties::CalculateMassAndInertia;
```

I am just wondering what is the reason that this happens? Is it because the inertia value is wrong for the given shape?
[Here is example code that can be placed into the convex hull test to replicate. A lot of it is auto generated so it is hard to look at. The code is inside the AddCustomStaticCompoundShape() function.](https://pastebin.com/LN1pCrGQ)

https://github.com/jrouwe/JoltPhysics/assets/39708313/a9f849cf-c044-46f9-99c0-1372449eaf39

**Comment by jrouwe** [comment, marked as answer]
Hello,

I think the main problem here is the inertia. If I change your code to:

```
creation_settings.mOverrideMassProperties = EOverrideMassProperties::CalculateInertia;
creation_settings.mMassPropertiesOverride.mMass = 1.0f;
```

Then it works without issues. The physics simulation sees an object with mass 1 and an inertia equivalent to the orange box:

![image](https://github.com/jrouwe/JoltPhysics/assets/1621693/4a666ac7-d30b-4e07-b273-023eafa0f026)

With your configuration, you also set the inertia to 1 meaning the simulation treats this house as if it was a small box:

![image](https://github.com/jrouwe/JoltPhysics/assets/1621693/e7f853da-e7ca-427e-a1f9-5904c13e8ea4)

The first contact manifold that it finds looks quite sane:

![image](https://github.com/jrouwe/JoltPhysics/assets/1621693/61a796da-0a20-40ef-924b-909354f04cf1)

But apparently it is not quite centered around the center of mass and because the inertia (= resistance to rotation) doesn't match the shape of the house it means it gets a much higher rotational speed that it should have so overshoots and becomes unstable.

In general I would recommend to let Jolt do the inertia calculation for you and only override the mass if need be.

**Reply by Pindrought** [reply]
Thanks so much as always!
